gui/mainwindow.cpp

```diff
@@ -79,11 +79,20 @@ MainWindow::MainWindow(TranslationHandler* th, QSettings* settings) :
     mLineEditFilter = new QLineEdit(mUI.mToolBarFilter);
     mLineEditFilter->setPlaceholderText(tr("Quick Filter:"));
     mUI.mToolBarFilter->addWidget(mLineEditFilter);
-    connect(mLineEditFilter, SIGNAL(textChanged(const QString&)), mFilterTimer, SLOT(start()));
+#ifndef QOVERLOAD_FALLBACK
+    connect(mLineEditFilter, &QLineEdit::textChanged, mFilterTimer, QOverload<>::of(&QTimer::start));
+#else
+    connect(mLineEditFilter, &QLineEdit::textChanged, mFilterTimer, static_cast<void (QTimer::*)(void)>(&QTimer::start));
+#endif
+
     connect(mLineEditFilter, &QLineEdit::returnPressed, this, &MainWindow::filterResults);
 
-    connect(mUI.mActionPrint, SIGNAL(triggered()), mUI.mResults, SLOT(print()));
-    connect(mUI.mActionPrintPreview, SIGNAL(triggered()), mUI.mResults, SLOT(printPreview()));
+#ifndef QOVERLOAD_FALLBACK
+    connect(mUI.mActionPrint, &QAction::triggered, mUI.mResults, QOverload<>::of(&ResultsView::print));
```

Comment:
Which cast do you mean? A signal with arguments and slot without? This was also so before only that the default parameter was not specified.

The only cast i used differently as before is the cast that say the compiler which of the overloaded ResultsView::print() methods I want to connect.

I think the 2 options you have named  are not practical in this case. The QAction have no other signal without arguments, that inform us about exactly the changes we need. The [void QAction::triggered(bool checked=false)](http://doc.qt.io/qt-5/qaction.html#triggered) is the only signal, that don't emit when setChecked() or toggle() is called. On the other hand, it would now be quite unnecessary to get an argument for all the functions, although they do not need it.

I do not know what added value it would bring to the functions ResultsView::printPreview(), MainWindow::close(), MainWindow::analyzeFiles(), etc., if the all are now given an argument.

The only thing which makes more sense here - as suggested by you above - lambda functions (as proxy) here. Should I adjust this everywhere, where the number of arguments differ from the signals and the slots?